### PR TITLE
Use sbt-web for asset resolving in facia end-to-end

### PR DESCRIPTION
**Pull request by rich-nguyen**
This fixes facia-end-to-end, at least the facia-tool side of things. It contains a few changes to facia-tool itself, to make the solution possible.

Just to be clear, I haven't suddenly decided to adopt [sbt-web](https://github.com/sbt/sbt-web) wholesale, it's an enabled plugin, and it's the primary way Play framework deals with the public asset folder. Whether we use it in conjunction with web jars to solve other issues in our assets pipeline remains unanswered. I think it has potential but we'd need a really compelling reason to choose to spend our efforts on a huge migration.

As you can see, I've re-jigged a few routes, and tweaked a few Asset controllers. The main cause of the problem is an issue which affects all our sbt multi projects: nowadays, sbt projects which depend on other projects communicate assets through packaged internal web jars. This means the Play framework classpath does not look the same as in previous versions. The upshot of this is that if you have a big project composed of little projects, if you need to use the assets from the little projects, you must do so through web target folder, which contains a lib folder.

@piuccio this look ok? shouldn't cause any problems...

UPDATE: Oh, one more thing. sbt-web provides a task called `assets` which you can run from facia-end-to-end in the sbt console. Running this task will ensure that your public-folder assets are put into the target folder.

### CODE REVIEW

File: project/Frontend.scala [hunk 1]
```diff
@@ -157,9 +157,11 @@ object Frontend extends Build with Prototypes {
     )
 
   val faciaEndToEnd = application("facia-end-to-end")
-    .dependsOn(commonWithTests)
-    .dependsOn(facia, faciaTool, faciaPress)
-    .aggregate(common, facia, faciaTool, faciaPress)
+    .dependsOn(faciaTool, facia, faciaPress)
+    .aggregate(faciaTool, facia, faciaPress)
+    .settings(
+      javaOptions in Runtime += "-Dconfig.file=facia-end-to-end/conf/facia-end-to-end.application.conf"
```
Comment (janua) [inline project/Frontend.scala]: You need this now? Why? What `conf` does it pickup without?

File: facia-end-to-end/conf/routes [hunk 2]
```diff
@@ -1,94 +1,97 @@
...
+GET         /login/status                                                   controllers.PandaAuthController.status
+
+# An sbt multi-project needs to route to internal web jars, kept in lib
+GET         /assets/facia-tool/*file                                        controllers.faciatool.UncachedAssets.at(file, relativePath="lib/facia-tool/")
 
 # static files
-GET           /humans.txt                controllers.Assets.at(path="/public", file="humans.txt")
-GET           /assets/*path              dev.DevAssetsController.at(path)
+GET         /humans.txt                                                     controllers.Assets.at(path="/public", file="humans.txt")
+GET         /assets/*file                                                   controllers.Assets.at(path="/public", file)
```
Comment (janua) [inline facia-end-to-end/conf/routes]: I wonder if this is used in `facia-end-to-end` at all?

Comment (rich-nguyen) [inline facia-end-to-end/conf/routes]: I did think that, yes.

**Comment by janua** [comment]
Hard to see the difference in routes. Need to use https://github.com/guardian/frontend/pull/9939/files?w=1.

It feels weird to me that they have introduced this problem without realising it. The fix seems like an obvious candidate for a major issue in `play`!
But maybe we're just too cowboy with our assets in the first place.

:+1:

File: project/Frontend.scala (hunk 1 above)
Comment (rich-nguyen) [inline project/Frontend.scala]: I believe it does a conf merge, thanks to the typesafe conf library. Given how similar the conf files are, this is usually benign, I just did it for completeness. I want us to move towards Play servers which only have a single router (`routes`), a single conf (`application.conf`), and a single `controllers.Assets` object

**Comment by rich-nguyen** [comment]
Well, end-to-end is a developer project to help preview the whole facia subsystem. It needn't be bullet-proof. I'm more concerned about Preview.

File: facia-end-to-end/conf/routes (hunk 2 above)
Comment (janua) [inline facia-end-to-end/conf/routes]: Actually, it's probably used by `facia`? Do all the paths to assets work properly on a facia page?

**Comment by piuccio** [comment]
Don't know why you need this, but as far as I'm concerned this looks good. :+1:

**Comment by rich-nguyen** [comment]
@piuccio it means that developers can run the facia-end-to-end project and still load assets in the public folder.

File: facia-end-to-end/conf/routes (hunk 2 above)
Comment (rich-nguyen) [inline facia-end-to-end/conf/routes]: Thanks, I'll make sure it's using the DevAssetsController. That won't affect the behaviour of facia-tool in dev

**Pull request merged**